_episodes_rmd/04-r-data-cleaning-merging.Rmd

```diff
@@ -31,151 +32,167 @@ knitr_fig_path("04-")
 > {: .solution}
 {: .challenge}
 
-The series column has two methods of quantifying CO2 emissions - "Emissions (thousand metric tons of carbon dioxide)" and "Emissions per capita (metric tons of carbon dioxide)". Those are long titles that we'd like to shorten to make them easier to work with. We can shorten them to "total_emissions" and "per_capita_emissions" using the recode function. We need to do this within the mutate function where we will mutate the series column. The syntax in the recode function is to tell recode which column we want to recode and then what the old value (e.g. "Emissions (thousand metric tons of carbon dioxide)") should equal after recoding (e.g. "total").
-
-```{r CleanEmissionsMutateRecode}
-co2_emissions_dirty %>% 
-  select(country, year, series, value) %>%
-  mutate(series = recode(series, "Emissions (thousand metric tons of carbon dioxide)" = "total_emissions",
-                         "Emissions per capita (metric tons of carbon dioxide)" = "per_capita_emissions"))
-```
 
-Recall that we'd like to have separate columns for the two ways that we CO2 emissions data. To achieve this, we'll use the pivot_wider function that we saw previously. The columns we want to spread out are series (i.e. names_from) and value (i.e. values_from).
+Another helpful thing to do when exploring and cleaning a new dataset is to get an idea of whether there are any missing values or duplicate rows. Let's start by checking to see if we have any **missing data**. We will do this using `drop_na()`, a tidyverse function that removes any rows that have missing values. We will then check the number of rows using `nrow` in our dataset and compare to the original to see if we lost any rows with missing data.
 
-```{r CleanEmissionsPivotWider}
-co2_emissions_dirty %>%
-  select(country, year, series, value) %>%
-  mutate(series = recode(series, "Emissions (thousand metric tons of carbon dioxide)" = "total_emission",
-                         "Emissions per capita (metric tons of carbon dioxide)" = "per_capita_emission")) %>%
-  pivot_wider(names_from=series, values_from=value)
+```{r drop_na}
+pollution_2019_dirty %>%
+  drop_na() %>%
+  nrow()
 ```
 
-Excellent! The last step before we can join this data frame is to get the most data that is for the year closest to 2007 so we can make a more direct comparison to the most recent data we have from gapminder. For the sake of time, we'll just tell you that we want data from 2005. 
+After dropping rows with missing values, our data object still has the same number of rows (690) as `pollution_2019_dirty`. This tells us that we don't have any rows with missing data. Wow! We're pretty lucky, because often datasets DO have missing data. What if we wanted to identify which observations had some missing values? Try the exercise below to find out!
 
-> ## Bonus: How did we determine that 2005 is the closest year to 2007? 
+> ## Filtering for missing data
```

Comment:
The way I was thinking of discussing missing data is to `full_join()` (or `left_join()`) the data (either all of it or just 2019) and then look for missing data in there. Maybe we can move this exercise to there because then there actually will be NAs?

Reply:
Done, but I made it a main point instead of an exercise.